src/Event/NotificationEvent.php

```diff
@@ -13,22 +13,37 @@
 
 class NotificationEvent extends ThemeEvent
 {
-    /**
-     * @var array<NotificationInterface>
-     */
+    public ?string $title = null;
+
+    public ?string $emptyTitle = null;
+
+    public bool $withArrow = true;
+
+    public string $badgeColor = 'red';
+
+    public bool $showBadgeTotal = true;
+
     private array $notifications = [];
 
+    public int $maxDisplay = 10;
+
     /**
-     * @param int|null $max
-     * @return NotificationInterface[]
+     * @return array<int,NotificationInterface>
      */
     public function getNotifications(?int $max = 10): array
     {
-        if (null !== $max) {
-            return \array_slice($this->notifications, 0, $max);
+        $notifications = array_filter(
+            $this->notifications,
+            fn(NotificationInterface $notification) => $notification->isValid()
```

Comment:
never used that syntax. which php version allows that ?

Reply:
It is a "js" way to make function.
Called ["Arrow function" from PHP 7.4](https://www.php.net/manual/en/functions.arrow.php)